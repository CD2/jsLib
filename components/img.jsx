import React from 'react'
import PropTypes from 'prop-types'

export default class Img extends React.Component {

  static propTypes = {
    alt: PropTypes.string
  }

  static defaultProps = {
    alt: '',
  }

  render() {
    return (
      <img alt={this.props.alt} {...this.props} />
    )
  }

}
