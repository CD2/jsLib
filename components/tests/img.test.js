import t from 'utils/test_helper'

import { Img } from '../img'

t.describe_component(Img, { alt: `test_alt` }, (c) => {
  c.accepts_prop(`alt`)

  c.has_css(`img`).with_attr(`alt`, `test_alt`)
})
