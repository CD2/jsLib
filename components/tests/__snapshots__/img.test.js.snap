// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Img /> renders correctly 1`] = `"<img alt=\\"\\">"`;
